AdminServer/appscale/admin/__init__.py

```diff
@@ -270,11 +273,25 @@ def initialize(self, acc, ua_client, zk_client, version_update_lock,
   def get(self):
     """ List projects.
     """
-    self.authenticate()
-    try:
-      projects = self.zk_client.get_children('/appscale/projects')
-    except NoNodeError:
-      projects = []
+    if 'AppScale-Secret' not in self.request.headers:
+      # Run all the authenticate functions in authenticate_access_token but
+      # don't check if user is authenticated for a specific project,
+      # get projects user can access.
+      token = self.request.headers['Authorization'].split()[1]
```

Comment:
This should probably handle the case when there is no "Authorization" header. A bad request status code would be more informative to the client than a 500.

Reply:
Bad Request or Unauthorized? In the base_handler if its missing that header or AppScale-Secret we say Unauthorized

Reply:
You're right. Unauthorized is more specific.